nautobot/extras/models/jobs.py

```diff
@@ -544,7 +545,9 @@ def enqueue_job(cls, func, name, obj_type, user, *args, celery_kwargs=None, sche
         schedule: Optional ScheduledJob instance to link to the JobResult
         kwargs: additional kwargs passed to the callable
         """
-        job_result = cls.objects.create(name=name, obj_type=obj_type, user=user, job_id=uuid.uuid4(), schedule=schedule)
+        job_result = cls.objects.create(
+            name=name, job_kwargs=kwargs, obj_type=obj_type, user=user, job_id=uuid.uuid4(), schedule=schedule
```

Comment:
Hi,

This was done this way originally to keep it in-line with the how the scheduledjob stores the job_kwarg data,
LMK if you want me to change it

Cheers,

Reply:
Fair enough - there's something to be said for self-consistency even if I think the extra data is mostly unnecessary. For reference:

```python
>>> pprint.pprint(ScheduledJob.objects.first().kwargs)
{'commit': True,
 'data': {'interval': 4},
 'name': 'plugins/example_plugin.jobs/ExampleLoggingJob',
 'request': {'GET': {'interval': '4'},
             'META': {'CONTENT_LENGTH': '974',
                      'CONTENT_TYPE': 'multipart/form-data; '
                                      'boundary=---------------------------4851323123551451491366783025',
                      'HTTP_ACCEPT': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8',
                      'HTTP_ACCEPT_ENCODING': 'gzip, deflate, br',
                      'HTTP_ACCEPT_LANGUAGE': 'en-US,en;q=0.5',
                      'HTTP_HOST': 'localhost:8080',
                      'HTTP_REFERER': 'http://localhost:8080/extras/jobs/plugins-example_plugin-jobs-exampleloggingjob/run/?interval=4',
                      'HTTP_USER_AGENT': 'Mozilla/5.0 (Macintosh; Intel Mac OS '
                                         'X 10.15; rv:101.0) Gecko/20100101 '
                                         'Firefox/101.0',
                      'QUERY_STRING': 'interval=4',
                      'REMOTE_ADDR': '172.20.0.1',
                      'REMOTE_HOST': '',
                      'REQUEST_METHOD': 'POST',
                      'SERVER_NAME': '207e539d08c4',
                      'SERVER_PORT': '8080'},
             'POST': {'_commit': 'on',
                      '_run': '',
                      '_schedule_name': 'Future!',
                      '_schedule_start_time': '2022-07-08 14:30:00',
                      '_schedule_type': 'future',
                      'csrfmiddlewaretoken': 'Ct11mYZTLOyg3bwjDlfCFd7qtSKhyVEuPye9SZs50Jq9fepCooaNHrQhVckyoraj',
                      'interval': '4'},
             '__nautobot_type__': 'nautobot.utilities.utils.NautobotFakeRequest',
             'id': 'bbd995f0-8745-46cf-9ef8-b85ba4c1f333',
             'path': '/extras/jobs/plugins-example_plugin-jobs-exampleloggingjob/run/',
             'user': '4f2dbb59-2a80-456b-9830-6b5f230d416a'},
 'scheduled_job_pk': 'dbc12c5b-46f6-4631-bd8d-2017cde5bdec',
 'user': '4f2dbb59-2a80-456b-9830-6b5f230d416a'}
```

```python
>>> pprint.pprint(JobResult.objects.first().job_kwargs)
{'commit': True,
 'data': {'interval': 4},
 'job_result_pk': 'c4487d9f-6537-4c56-96fe-19bc5210ce01',
 'request': {'GET': {},
             'META': {'CONTENT_LENGTH': '953',
                      'CONTENT_TYPE': 'multipart/form-data; '
                                      'boundary=---------------------------1997628343772709020923472054',
                      'HTTP_ACCEPT': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8',
                      'HTTP_ACCEPT_ENCODING': 'gzip, deflate, br',
                      'HTTP_ACCEPT_LANGUAGE': 'en-US,en;q=0.5',
                      'HTTP_HOST': 'localhost:8080',
                      'HTTP_REFERER': 'http://localhost:8080/extras/jobs/plugins-example_plugin-jobs-exampleloggingjob/run/',
                      'HTTP_USER_AGENT': 'Mozilla/5.0 (Macintosh; Intel Mac OS '
                                         'X 10.15; rv:101.0) Gecko/20100101 '
                                         'Firefox/101.0',
                      'QUERY_STRING': '',
                      'REMOTE_ADDR': '172.20.0.1',
                      'REMOTE_HOST': '',
                      'REQUEST_METHOD': 'POST',
                      'SERVER_NAME': '207e539d08c4',
                      'SERVER_PORT': '8080'},
             'POST': {'_commit': 'on',
                      '_run': '',
                      '_schedule_name': '',
                      '_schedule_start_time': '',
                      '_schedule_type': 'immediately',
                      'csrfmiddlewaretoken': 'yQJCyf5E7hO1vO2BA0mfg53M0c7Q3JHiLVWK4gyQmcGUHRVUl3hqijMDswH7Tfd7',
                      'interval': '4'},
             '__nautobot_type__': 'nautobot.utilities.utils.NautobotFakeRequest',
             'id': 'c492ee45-26ed-49ca-aa52-af854c04f414',
             'path': '/extras/jobs/plugins-example_plugin-jobs-exampleloggingjob/run/',
             'user': '4f2dbb59-2a80-456b-9830-6b5f230d416a'}}
```

Reply:
I want to revisit this as it makes the REST API for JobResults a bit ugly with all of the extraneous data present in job_kwargs. Plus it was causing some test failures, for which I've implemented a workaround in b516dc0, but I'm not entirely happy with that workaround. It will probably be Monday before I can take another look at this though.

Reply:
Ahh insee there are a few differences, would you prefer it if i just update it only contain the "data"

Reply:
I'm not 100% sure on this - it feels like we should probably also convey the `commit` flag (which isn't mapped currently to the "run" page, as it's not part of the `data`, but a separate field) as well, but I'm not sure. Would also appreciate some thoughts from @jathanism when he has time.